src/Symfony/Component/Translation/Translator.php

```diff
@@ -376,25 +372,37 @@ private function getFallbackContent(MessageCatalogue $catalogue)
             $fallbackSuffix = ucfirst(preg_replace($replacementPattern, '_', $fallback));
             $currentSuffix = ucfirst(preg_replace($replacementPattern, '_', $current));
 
-            $fallbackContent .= sprintf(<<<EOF
-\$catalogue%s = new MessageCatalogue('%s', %s);
-\$catalogue%s->addFallbackCatalogue(\$catalogue%s);
-
-EOF
-                ,
-                $fallbackSuffix,
-                $fallback,
-                var_export($fallbackCatalogue->all(), true),
-                $currentSuffix,
-                $fallbackSuffix
-            );
+            $fallbackContent = $this->getCataloguePhp($fallback, $fallbackCatalogue->all(), $fallbackSuffix);
+            $fallbackContent .= sprintf("\$catalogue%s->addFallbackCatalogue(\$catalogue%s);\n", $currentSuffix, $fallbackSuffix);
+
             $current = $fallbackCatalogue->getLocale();
             $fallbackCatalogue = $fallbackCatalogue->getFallbackCatalogue();
         }
 
         return $fallbackContent;
     }
 
+    private function getCataloguePhp($locale, $messages, $suffix)
+    {
+        $static = 'static ';
+        foreach ($messages as $msgs) {
+            if (count($msgs) > 32767) {
+                $static = '';
+                break;
+            }
+        }
+
+        $messages = str_replace("' . \"\\0\" . '", "\0", var_export($messages, true));
```

Comment:
Prevent ant at compile time string concatenation requirement for.PHP 5.5 compat: `var_export` generates such things (and no others in the current case).